docs/configuration/service/config-sync.rst

```diff
@@ -0,0 +1,117 @@
+.. _config-sync:
+
+###########
+Config Sync
+###########
+
+Configuration synchronization (config sync) is a feature of VyOS that 
+permits synchronization of the configuration of one VyOS router to 
+another in a network. 
+
+The main benefit to configuration synchronization is that it eliminates having  
+to manually replicate configuration changes made on the primary router to the  
+secondary (replica) router.
+
+The writing of the configuration to the secondary router is performed through 
+the VyOS HTTP API. The user can specify which portion(s) of the configuration will 
+be synchronized and the mode to use - whether to replace or add. 
+
+To prevent issues with divergent configurations between the pair of routers, 
+synchronization is strictly unidirectional from primary to replica. Both 
+routers should be online and run the same version of VyOS.
+
+Configuration
+-------------
+
+.. cfgcmd:: set service config-sync section secondary 
+   <address|key|timeout|port>
+
+   Specify the address, API key, timeout and port of the secondary router. 
+   You need to enable and configure the HTTP API service on the secondary 
+   router for config sync to operate.
+   
+.. cfgcmd:: set service config-sync section <section>
+
+   Specify the section of the configuration to synchronize. If more than one 
+   section is to be synchronized, repeat the command to add additional 
+   sections as required.
+
+.. cfgcmd:: set service config-sync mode <load|set>
+
+   Two options are available for `mode`: either `load` and replace or `set`
+   the configuration section.
+
+.. code-block:: none
+
+    Supported options for <section> include:
+        firewall
+        interfaces <interface>
+        nat
+        nat66
+        pki
+        policy
+        protocols <protocol>
+        qos <interface|policy>
+        service <service>
+        system <conntrack| 
+        flow-accounting|option|sflow|static-host-mapping|sysctl|time-zone>
+        vpn
+        vrf
+
+Example
+-------
+* Synchronize the time-zone and OSPF configuration from Router A to Router B
+* The address of Router B is 10.0.20.112 and the port used is 8443
+
+Configure the HTTP API service on Router B
+
+.. code-block:: none
+
+    set service https api graphql authentication expiration '31536000'
```

Comment:
Graphql is not required

Reply:
Thank you, will omit and retest without to ensure.

Reply:
@sever-sever confirming it works with just a basic HTTPS-API config see below. I've modified the documentation as you suggested.
`
vyos@vyos-B# delete service https api graphql 
vyos@vyos-B# commit
vyos@vyos-B# run show configuration commands | match "https"
set service https api keys id KID key 'foo'
set service https listen-address '10.0.20.113'
set service https port '8443'
`
**Router A**
% ssh vyos@10.0.20.111
vyos@10.0.20.111's password: 
Welcome to VyOS!

   ┌── ┐
   . VyOS 1.5-rolling-202405060019
   └ ──┘  current

 * Documentation:  https://docs.vyos.io/en/latest
 * Project news:   https://blog.vyos.io
 * Bug reports:    https://vyos.dev

You can change this banner using "set system login banner post-login" command.

VyOS is a free software distribution that includes multiple components,
you can check individual component licenses under /usr/share/doc/*/copyright
Last login: Tue May  7 16:42:28 2024 from 10.0.20.71
vyos@vyos-A:~$ conf
vyos@vyos-A# set system time-zone 'Europe/Amsterdam'
vyos@vyos-A# commit
INFO:vyos_config_sync:Config synchronization: Mode=load, Secondary=10.0.20.113
vyos@vyos-A# set protocols ospf area 0 network '10.111.222.0/30'
vyos@vyos-A# commit
INFO:vyos_config_sync:Config synchronization: Mode=load, Secondary=10.0.20.113

**Router B**
% ssh vyos@10.0.20.113
vyos@10.0.20.113's password: 
Welcome to VyOS!

   ┌── ┐
   . VyOS 1.5-rolling-202405060019
   └ ──┘  current

 * Documentation:  https://docs.vyos.io/en/latest
 * Project news:   https://blog.vyos.io
 * Bug reports:    https://vyos.dev

You can change this banner using "set system login banner post-login" command.

VyOS is a free software distribution that includes multiple components,
you can check individual component licenses under /usr/share/doc/*/copyright
Last login: Tue May  7 13:35:23 2024 from 10.0.20.71
vyos@vyos-B:~$ show configuration commands | match "time-zone"
set system time-zone **'Europe/Amsterdam'**
vyos@vyos-B:~$ show configuration commands | match "ospf"
set protocols ospf area 0 network '192.0.2.0/30'
set protocols ospf area 0 network **'10.111.222.0/30'**